src/generic/numberparsing.h

```diff
@@ -1134,6 +1134,23 @@ static bool parse_float_strtod(const char *ptr, double *outDouble) {
   // Some libraries will set errno = ERANGE when the value is subnormal,
   // yet we may want to be able to parse subnormal values.
   // However, we do not want to tolerate NAN or infinite values.
+  //
+  // Values like infinity or NaN are not allowed in the JSON specification.
```

Comment:
RapidJSON supports non-standard values like NaN and Inf. Similarly, environments such as MATLAB, Octave, Scilab, and Nelson, which use C++ libraries for JSON parsing also accept these values. For example: jsondecode('NaN') works as expected. While this behavior is not part of the official JSON standard, it is commonly accepted in legacy code and widely supported in practice...

Reply:
The resulting pseudo-JSON won't be parsed in Python and JavaScript.

<img width="543" height="99" alt="Capture d’écran, le 2025-07-30 à 20 14 41" src="https://github.com/user-attachments/assets/dac2a39c-cca2-4867-9d97-fe8b3654b2c9" />
<img width="570" height="371" alt="Capture d’écran, le 2025-07-30 à 20 14 50" src="https://github.com/user-attachments/assets/97d794f1-468d-4254-b9d3-a5bbf0d3e17b" />

It seems much more sensible to fix the JSON generators.

In simdjson, we do support (in an undocumented manner) JSON variants (hidden behind a macro). If you need such an extension, I encourage you to produce a pull request.

Reply:
It could be quite useful to support additional parsing options, even if they're not strictly standard, similar to fast_float::parse_options options { fast_float::chars_format::json_or_infnan }; :)